src/Elliptic/SubdomainPreconditioners/MinusLaplacian.hpp

```diff
@@ -0,0 +1,256 @@
+              ::LinearSolver::Serial::Registrars::Gmres<
+                  ::LinearSolver::Schwarz::ElementCenteredSubdomainData<
+                      Dim, tmpl::list<Poisson::Tags::Field>>>,
+              ::LinearSolver::Serial::Registrars::ExplicitInverse>>>
+struct MinusLaplacian {
+  template <typename LinearSolverRegistrars>
+  using f = subdomain_preconditioners::MinusLaplacian<Dim, OptionsGroup, Solver,
+                                                      LinearSolverRegistrars>;
+};
+}  // namespace Registrars
+
+/*!
+ * \brief Approximate the subdomain operator with a flat-space Laplacian with
+ * Dirichlet boundary conditions for every tensor component separately.
+ *
+ * This linear solver applies the `Solver` to every tensor component in
+ * turn, approximating the subdomain operator with a flat-space Laplacian with
+ * Dirichlet boundary conditions. This can be a lot cheaper than solving the
+ * full subdomain operator and can provide effective preconditioning for an
+ * iterative subdomain solver. The approximation is better the closer the
+ * original PDEs are to a set of decoupled flat-space Poisson equations with
+ * Dirichlet boundary conditions.
+ *
+ * \tparam Dim Spatial dimension
+ * \tparam OptionsGroup The options group identifying the
+ * `LinearSolver::Schwarz::Schwarz` solver that defines the subdomain geometry.
+ * \tparam Solver Any class that provides a `solve` and a `reset` function,
+ * but typically a `LinearSolver::Serial::LinearSolver`. The solver will be
+ * factory-created from input-file options.
+ */
+template <size_t Dim, typename OptionsGroup,
+          typename Solver = LinearSolver::Serial::LinearSolver<tmpl::list<
+              ::LinearSolver::Serial::Registrars::Gmres<
+                  ::LinearSolver::Schwarz::ElementCenteredSubdomainData<
+                      Dim, tmpl::list<Poisson::Tags::Field>>>,
+              ::LinearSolver::Serial::Registrars::ExplicitInverse>>,
+          typename LinearSolverRegistrars =
+              tmpl::list<Registrars::MinusLaplacian<Dim, OptionsGroup, Solver>>>
+class MinusLaplacian
+    : public LinearSolver::Serial::LinearSolver<LinearSolverRegistrars> {
+ private:
+  using Base = LinearSolver::Serial::LinearSolver<LinearSolverRegistrars>;
+  using SolverType = tmpl::conditional_t<std::is_abstract_v<Solver>,
+                                         std::unique_ptr<Solver>, Solver>;
+
+ public:
+  static constexpr size_t volume_dim = Dim;
+  using options_group = OptionsGroup;
+  using poisson_system =
+      Poisson::FirstOrderSystem<Dim, Poisson::Geometry::FlatCartesian>;
+  using SubdomainOperator =
+      elliptic::dg::subdomain_operator::SubdomainOperator<poisson_system,
+                                                          OptionsGroup>;
+  using SubdomainData = ::LinearSolver::Schwarz::ElementCenteredSubdomainData<
+      Dim, tmpl::list<Poisson::Tags::Field>>;
+  using solver_type = Solver;
```

Comment:
You have type aliases `SolverType` and `solver_type` that are different things.  This seems confusing.  Can you rename one of them?